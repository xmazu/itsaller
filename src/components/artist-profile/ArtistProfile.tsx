import React from 'react';

import { Artist } from '../../types';
import './artist-profile.scss';
import Icon from '../icon';

export interface ArtistProfileProps {
  artist: Artist;
}

export default class ArtistProfile extends React.Component<ArtistProfileProps> {
  render() {
    const { artist } = this.props;
    return (
      <div className="artistProfile">
        <div className="artistProfile__image">
          <img src={artist.image} alt={artist.name} />
        </div>
        <div className="artistProfile__name">
          <h1>{artist.name}</h1>
        </div>
        <div className="artistProfile__link">
          <a href={artist.facebook} target="_blank">
            Facebook profile
          </a>
        </div>
      </div>
    );
  }
}
