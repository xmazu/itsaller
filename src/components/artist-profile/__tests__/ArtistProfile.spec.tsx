import React from 'react';
import { shallow } from 'enzyme';

import ArtistProfile from '../ArtistProfile';

describe('<ArtistProfile />', () => {
  it('renders correctly', () => {
    const mockedEvent = {
      date: "2018-12-30T19:00:48",
      city: 'Krakow',
      country: 'Poland',
      venue: 'Tauron Arena',
      lat: 0,
      long: 0
    };
    const artist = {
      facebook: 'http://fb.me',
      image: 'https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg',
      name: 'Catzilla'
    };
    const component = shallow(
      <ArtistProfile events={[mockedEvent]} artist={artist} />
    );
    expect(component).toMatchSnapshot();
  });
});
