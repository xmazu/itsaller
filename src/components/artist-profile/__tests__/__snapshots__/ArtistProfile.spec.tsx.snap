// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistProfile /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArtistProfile
    artist={
      Object {
        "facebook": "http://fb.me",
        "image": "https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg",
        "name": "Catzilla",
      }
    }
    events={
      Array [
        Object {
          "city": "Krakow",
          "country": "Poland",
          "date": "2018-12-30T19:00:48",
          "id": "my.id",
          "lat": 0,
          "long": 0,
          "venue": "Tauron Arena",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="artistProfile__details"
        >
          <div
            className="artistProfile__image"
          >
            <img
              alt="Catzilla"
              src="https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg"
            />
          </div>
          <div
            className="artistProfile__name"
          >
            <h1>
              Catzilla
            </h1>
          </div>
          <div
            className="artistProfile__link"
          >
            <a
              href="http://fb.me"
              target="_blank"
            >
              Facebook profile
            </a>
          </div>
        </div>,
        <div
          className="artistProfile_events"
        >
          <EventsList
            events={
              Array [
                Object {
                  "city": "Krakow",
                  "country": "Poland",
                  "date": "2018-12-30T19:00:48",
                  "id": "my.id",
                  "lat": 0,
                  "long": 0,
                  "venue": "Tauron Arena",
                },
              ]
            }
          />
        </div>,
      ],
      "className": "artistProfile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="artistProfile__image"
            >
              <img
                alt="Catzilla"
                src="https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg"
              />
            </div>,
            <div
              className="artistProfile__name"
            >
              <h1>
                Catzilla
              </h1>
            </div>,
            <div
              className="artistProfile__link"
            >
              <a
                href="http://fb.me"
                target="_blank"
              >
                Facebook profile
              </a>
            </div>,
          ],
          "className": "artistProfile__details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Catzilla"
                src="https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg"
              />,
              "className": "artistProfile__image",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Catzilla",
                "src": "https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                Catzilla
              </h1>,
              "className": "artistProfile__name",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Catzilla",
              },
              "ref": null,
              "rendered": "Catzilla",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                href="http://fb.me"
                target="_blank"
              >
                Facebook profile
              </a>,
              "className": "artistProfile__link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Facebook profile",
                "href": "http://fb.me",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Facebook profile",
              "type": "a",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <EventsList
            events={
              Array [
                Object {
                  "city": "Krakow",
                  "country": "Poland",
                  "date": "2018-12-30T19:00:48",
                  "id": "my.id",
                  "lat": 0,
                  "long": 0,
                  "venue": "Tauron Arena",
                },
              ]
            }
          />,
          "className": "artistProfile_events",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "events": Array [
              Object {
                "city": "Krakow",
                "country": "Poland",
                "date": "2018-12-30T19:00:48",
                "id": "my.id",
                "lat": 0,
                "long": 0,
                "venue": "Tauron Arena",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="artistProfile__details"
          >
            <div
              className="artistProfile__image"
            >
              <img
                alt="Catzilla"
                src="https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg"
              />
            </div>
            <div
              className="artistProfile__name"
            >
              <h1>
                Catzilla
              </h1>
            </div>
            <div
              className="artistProfile__link"
            >
              <a
                href="http://fb.me"
                target="_blank"
              >
                Facebook profile
              </a>
            </div>
          </div>,
          <div
            className="artistProfile_events"
          >
            <EventsList
              events={
                Array [
                  Object {
                    "city": "Krakow",
                    "country": "Poland",
                    "date": "2018-12-30T19:00:48",
                    "id": "my.id",
                    "lat": 0,
                    "long": 0,
                    "venue": "Tauron Arena",
                  },
                ]
              }
            />
          </div>,
        ],
        "className": "artistProfile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="artistProfile__image"
              >
                <img
                  alt="Catzilla"
                  src="https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg"
                />
              </div>,
              <div
                className="artistProfile__name"
              >
                <h1>
                  Catzilla
                </h1>
              </div>,
              <div
                className="artistProfile__link"
              >
                <a
                  href="http://fb.me"
                  target="_blank"
                >
                  Facebook profile
                </a>
              </div>,
            ],
            "className": "artistProfile__details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Catzilla"
                  src="https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg"
                />,
                "className": "artistProfile__image",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Catzilla",
                  "src": "https://i.ytimg.com/vi/5530I_pYjbo/maxresdefault.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  Catzilla
                </h1>,
                "className": "artistProfile__name",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Catzilla",
                },
                "ref": null,
                "rendered": "Catzilla",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  href="http://fb.me"
                  target="_blank"
                >
                  Facebook profile
                </a>,
                "className": "artistProfile__link",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Facebook profile",
                  "href": "http://fb.me",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Facebook profile",
                "type": "a",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <EventsList
              events={
                Array [
                  Object {
                    "city": "Krakow",
                    "country": "Poland",
                    "date": "2018-12-30T19:00:48",
                    "id": "my.id",
                    "lat": 0,
                    "long": 0,
                    "venue": "Tauron Arena",
                  },
                ]
              }
            />,
            "className": "artistProfile_events",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "events": Array [
                Object {
                  "city": "Krakow",
                  "country": "Poland",
                  "date": "2018-12-30T19:00:48",
                  "id": "my.id",
                  "lat": 0,
                  "long": 0,
                  "venue": "Tauron Arena",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
